preview-generator.ts

```diff
@@ -21,48 +22,44 @@ const drawSide = async (side: 'left' | 'right', ctx: any, image: Image, question
   ctx.drawImage(image, x, y, newWidth, newHeight)
   drawImageProp(ctx, image, x, 0, width / 2, height, 0.5, 0.5)
   createTextWithBackground(
-    side === 'left' ? question.choiceLeft.title.toUpperCase() : question.choiceRight.title.toUpperCase(),
+    choice.title.toUpperCase(),
     x,
     ctx,
     width,
     height
   )
 }
 
-const renderPreviewBySlug = async (slug: string, override = false) => {
-  const question = questions.find((q) => q.slug === slug)
-  if (!question) return
+const renderPreview = async (question: QuestionInterface, override = false) => {
 
-  if (fs.existsSync(`${assetPath}/assets/img-previews/preview-${question.slug}.jpg`) && !override) {
+  if (!override && existsSync(`${assetPath}/assets/img-previews/preview-${question.slug}.jpg`)) {
     console.log(`Preview for ${question.slug} already exists`)
     return
   }
 
-  if (question.choiceLeft.img_path.indexOf('webp') > -1 || question.choiceRight.img_path.indexOf('webp') > -1) {
+  if (question.choiceLeft.img_path.endsWith('.webp') || question.choiceRight.img_path.endsWith('webp')) {
     console.log('Webp not supported')
     return
   }
 
-  const promiseImg1 = loadImage(assetPath + question.choiceLeft.img_path)
-  const promiseImg2 = loadImage(assetPath + question.choiceRight.img_path)
-  Promise.all([promiseImg1, promiseImg2]).then(async (images) => {
+  const leftImagePromise = loadImage(assetPath + question.choiceLeft.img_path)
+  const rightImagePromise = loadImage(assetPath + question.choiceRight.img_path)
+  Promise.all([leftImagePromise, rightImagePromise]).then(async ([ leftImage, rightImage ]) => {
     const canvas = new Canvas(width, height)
     const ctx = canvas.getContext('2d')
 
-    await drawSide('left', ctx, images[0], question)
-    await drawSide('right', ctx, images[1], question)
+    await drawSide('left', ctx, leftImage, question.choiceLeft)
+    await drawSide('right', ctx, rightImage, question.choiceRight)
 
-    const out = fs.createWriteStream(__dirname + `/${assetPath}/assets/img-previews/preview-${slug}.jpg`)
+    const out = createWriteStream(`${assetPath}/assets/img-previews/preview-${question.slug}.jpg`)
     const stream = canvas.createJPEGStream({
       quality: 0.8,
     })
     stream.pipe(out)
-    out.on('finish', () => console.log('The JPG preview-' + slug + '.jpg file was created.'))
+    out.on('finish', () => console.log(`The JPG preview-${question.slug}.jpg file was created.`))
   })
 }
 
-//renderPreviewBySlug('camelCase-or-snake_case', true).then(() => {})
-
-questions.forEach(async (question) => {
-  await renderPreviewBySlug(question.slug, override)
+questions.forEach((question) => {
+  renderPreview(question, override)
```

Comment:
we don't have things to do after `renderPreview` and node will always wait for the entire code to run. So no need of `await`